x-pack/plugins/observability_solution/infra/server/saved_objects/custom_dashboards/custom_dashboards_saved_object.ts

```diff
@@ -13,12 +13,29 @@ import { InfraCustomDashboard } from '../../../common/custom_dashboards';
 export const INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE = 'infra-custom-dashboards';
 
 const properties: Record<keyof InfraCustomDashboard, SavedObjectsFieldMapping> = {
-  dashboardIdList: { type: 'keyword' },
-  assetType: { type: 'keyword' },
-  kuery: { type: 'text' },
+  assetType: {
+    type: 'keyword',
+  },
+  dashboardIdList: {
+    properties: {
+      id: {
+        type: 'keyword',
+      },
+      hostNameFilterEnabled: {
+        type: 'keyword',
+      },
+    },
+    type: 'nested',
+  },
+  kuery: {
+    type: 'text',
+  },
```

Comment:
I don't see  the added mappings declared in a corresponding new model version on the `infraCustomDashboardsSavedObjectType` definition in the same file. 

Was the mapping check script ran manually to make the CI happy (given I see the mappings still added in the `current_fields` and `current_mappings`?)

Reply:
Note: I want to replace the version 1 schema as the object is not used so I assumed that replacing it would be enough but maybe I am missing something 🤔

Reply:
> but it's a relatively new saved object and I wasn't sure if we needed to create a new model version

You always do, unless the SO type is introduced in the same PR

>  in this case we need to define it as model version 2 and keep the old version as well

That's correct. Any mapping changes require the addition of a new MV, to keep track of the incremental changes for the zero downtime upgrades for serverless.

Reply:
@jennypavlova this makes sense to me, creating another version when mappings changed.  

A few other things about this saved object (@mykolaharmash @crespocarlos): 

-  If it necessary to map all these fields?  If I am understanding correctly, we are only querying on the assetType field
- we should remove `kuery` field if we are not using it.  I don't see how it makes sense in this design to filter on every host.  what's the use case?  It makes sense in the APM design of creating a saved object per dashboard link.  
- any reason not to add `dynamic:false` to the mappings?  That way we don't automatically index any field.

Reply:
Thank you @neptunian and @pgayvallet! I updated the saved object with `V2` schema, removed `kuery` (not used currently), and added `dynamic:false`. Can you please take a look?
The mappings can help in case we extend the API to filter the dashboards based on a flag or other properties inside `dashboardIdList` - do we want to have only `assetType` inside the `properties` and extend it later in case we query other fields?